public/src/components/forms/GroupForm.tsx

```diff
@@ -253,16 +253,19 @@ export class GroupForm extends React.Component<IGroupProps, IGroupState> {
 
         if (!this.userValidate(this.state.curUser)) {
             errors.push("You must be a member of the group");
-        }        
+        }
         return errors;
     }
 
     private userValidate(curUser: IUserRelation | undefined): boolean {
-        if (curUser && curUser.link.getStatus() === Enrollment.UserStatus.TEACHER) {
-            return true
-        } else {
-            return curUser ? curUser.link.getStatus() === Enrollment.UserStatus.STUDENT && this.isCurrentStudentSelected(curUser) : false
+        if (!curUser) {
+            return false;
         }
+        if (!((curUser.enrollment.getStatus() === Enrollment.UserStatus.TEACHER) ||
+         (curUser.enrollment.getStatus() === Enrollment.UserStatus.STUDENT && this.isCurrentStudentSelected(curUser)))) {
```

Comment:
Hmmm. I was expecting this logic to become simpler too. If I understand the logic right here, what you are testing is:
```typescript
const status = curUser.enrollment.getStatus()
return status === Enrollment.UserStatus.TEACHER || (status === Enrollment.UserStatus.STUDENT && this.isCurrentStudentSelected(curUser))
```
(there is no need for an if-statement here)

Another alternative could be make this into a method on `this`, similar to the `isCurrentStudentSelected` method, or better yet make these as methods on the `IUserRelation` object, so that you avoid the `this` object.

Reply:
Oh, right, got it now. Fixed.